pufferlib/ocean/trash_pickup/trash_pickup.h

```diff
@@ -0,0 +1,596 @@
+#ifndef TRASH_PICKUP_H
+#define TRASH_PICKUP_H
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <stdio.h>
+#include "raylib.h"
+
+#define EMPTY 0
+#define TRASH 1
+#define TRASH_BIN 2
+#define AGENT 3
+
+#define ACTION_UP 0
+#define ACTION_DOWN 1
+#define ACTION_LEFT 2
+#define ACTION_RIGHT 3
+
+#define AGENT_IS_CARRYING 1
+#define AGENT_IS_NOT_CARRYING 0
+
+#define LOG_BUFFER_SIZE 1024
+
+// Helper macro for 1D indexing
+#define INDEX(env, x, y) ((y) * (env)->grid_size + (x))
+
+typedef struct Log {
+    float episode_return;
+    float episode_length;
+    float trash_collected;
+} Log;
+
+typedef struct LogBuffer {
+    Log* logs;
+    int length;
+    int idx;
+} LogBuffer;
+
+// LogBuffer functions
+LogBuffer* allocate_logbuffer(int size) {
+    LogBuffer* logs = (LogBuffer*)calloc(1, sizeof(LogBuffer));
+    logs->logs = (Log*)calloc(size, sizeof(Log));
+    logs->length = size;
+    logs->idx = 0;
+    return logs;
+}
+
+void free_logbuffer(LogBuffer* buffer) {
+    free(buffer->logs);
+    free(buffer);
+}
+
+Log aggregate_and_clear(LogBuffer* logs) {
+    Log log = {0};
+    if (logs->idx == 0) {
+        return log;
+    }
+    for (int i = 0; i < logs->idx; i++) {
+        log.episode_return += logs->logs[i].episode_return;
+        log.episode_length += logs->logs[i].episode_length;
+        log.trash_collected += logs->logs[i].trash_collected;
+    }
+    log.episode_return /= logs->idx;
+    log.episode_length /= logs->idx;
+    log.trash_collected /= logs->idx;
+    logs->idx = 0;
+    return log;
+}
+
+void add_log(LogBuffer* logs, Log* log) {
+    if (logs->idx == logs->length) {
+        return;
+    }
+    logs->logs[logs->idx] = *log;
+    logs->idx += 1;
+}
+
+typedef struct {
+    int type;      // Entity type: EMPTY, TRASH, TRASH_BIN, AGENT
+    int index;     // Index in the positions array (-1 if not applicable)
+} GridCell;
+
+typedef struct {
+    // Interface for PufferLib
+    float* observations;
+    int* actions;
+    float* rewards;
+    unsigned char* dones;
+    LogBuffer* log_buffer;
+
+    int grid_size;
+    int num_agents;
+    int num_trash;
+    int num_bins;
+    int max_steps;
+    int current_step;
+
+    float positive_reward;
+    float negative_reward;
+    float total_episode_reward;
+
+    GridCell* grid; // 1D array for grid
+    int (*agent_positions)[2]; // Agent positions
```

Comment:
I would make an Entity struct with type (agent/bin/trash) and position

Reply:
Replaced with Entity struct (to be pushed).